src/app/Library/CrudPanel/Traits/Fields.php

```diff
@@ -35,6 +35,17 @@ public function addField($field)
             $newField = $field;
         }
 
+        //we set this up exclusive for relationship type field. in future we hope we can figure out this automatically from field name
+        //atm we avoid any breaking changes while developing the guessing habilities for crud panel
+        if (isset($newField['type']) && $newField['type'] == 'relationship') {
+            $relationData = $this->getRelationWithFieldName($newField['name']);
```

Comment:
Agree with you in renaming to getRelationFromFieldName(). - fixed

The field name can be the relation name **articles** or the column name **article_id** - i created the functionality, but i have mixed feelings about having this here. It's more code to maintain, and well if it's a relationship field you should provide the direct relation name. I am willing to remove this feature if you share the same concerns as i do. 

Do you see why would people use array as field name for a relationship field? Is it intended for example:
```php
$this->crud->addField(['name' => ['relation1','relation2'],'type' => 'relationship']);

//instead of:

$this->crud->addField(['name' => relation1','type' => 'relationship']);
$this->crud->addField(['name' => relation2','type' => 'relationship']);
```
Adds a field for each relation ?

Reply:
I don't see people using ```name``` arrays for the relationship field, no. I don't think that's worth supporting. Don't know why I mentioned that earlier, if we don't support it - we don't support it 😄  nothing we should do about it.

----

Hmm... you're right - nice catch with being able to put both ```category``` and ```category_id``` for the name, and be able to infer it's a relationship field. BUT. Wouldn't that interfere with the saving process? As far as I know, when saving:
- for 1-n relationship, the ```name``` has to be ```category_id``` (column in db)
- for n-n relationship, the ```name``` should be ```categories``` (relationship name) and ```pivot``` has to be ```true```;

The way I understand this change, this would make it possible for 1-n relations to put both ```category_id``` and ```category``` for name - which would make it consistent between relationships, which is GREAT. But only if the saving process still works 😄